messaging/src/main/java/org/axonframework/eventhandling/pooled/PooledStreamingEventProcessor.java

```diff
@@ -672,13 +673,13 @@ public Builder tokenClaimInterval(long tokenClaimInterval) {
          * Defines the maximum number of segment this {@link StreamingEventProcessor} may claim. Defaults to {@value
          * Short#MAX_VALUE}.
          *
-         * @param maxClaimedSegments the maximum number fo claimed segments for this {@link StreamingEventProcessor}
+         * @param maxSegmentProvider the maximum number fo claimed segments for this {@link StreamingEventProcessor}
          *
          * @return the current Builder instance, for fluent interfacing
          */
-        public Builder maxClaimedSegments(int maxClaimedSegments) {
-            assertStrictPositive(maxClaimedSegments, "Max claimed segments should be a higher valuer than zero");
-            this.maxClaimedSegments = maxClaimedSegments;
+        public Builder maxSegmentProvider(MaxSegmentProvider maxSegmentProvider) {
+            assertStrictPositive(maxSegmentProvider.getMaxSegments(name), "Max claimed segments should be a higher valuer than zero");
```

Comment:
I would prefer to have an `assertNonNull` before the `assertStrictPositive` to have a better message in case it's used with null.

Reply:
done.